lib/client.js

```diff
@@ -217,10 +217,15 @@ class Client extends EventEmitter {
       // TODO move this[kCallbacks] from being an array. The array allocation
       // is showing up in the flamegraph.
       const cb = this[kCallbacks].shift()
+      const request = this[kRequests].shift()
+      let skipBody
       this._needHeaders--
-      this._lastBody = new Readable({ read: this[kRead].bind(this) })
-      this._lastBody.push = this[kRequests].shift().wrapSimple(this._lastBody, this._lastBody.push)
-
+      if (request.method === 'HEAD') {
+        skipBody = 1
```

Comment:
So by returning `1` or `true` the parser finishes all the requests

Reply:
Could this be node version dependent?
https://github.com/creationix/http-parser-js/blob/8a81c92201e49fda4668386263fdce284789ed73/http-parser.js#L362

Reply:
I don’t know. why do you ask?

Reply:
Because of this comment in the snippet I posted
```js
      // For older versions of node (v6.x and older?), that return skipBody=1 or skipBody=true,
      //   need this "return true;" if it's an upgrade request.
```